src/formatDate.js

```diff
@@ -9,6 +41,31 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  let datePartsMatched = {};
+
+  for (let i = 0; i < oldDateParts.length; i++) {
+    datePartsMatched[fromFormat[i]] = oldDateParts[i];
+  }
+
+  datePartsMatched = convertYear(datePartsMatched, fromFormat, toFormat);
```

Comment:
The `convertYear` function is modifying the `datePartsMatched` object directly. While this works, it's generally good practice to avoid side effects where possible. Consider returning a new object with the updated year instead of modifying the input object.